pkg/entrypoint/admissioncontroller/admissioncontroller_test.go

```diff
@@ -0,0 +1,1084 @@
+			response: &admissionv1beta1.AdmissionResponse{
+				UID:     "uid",
+				Allowed: true,
+				Result: &metav1.Status{
+					Status: metav1.StatusSuccess,
+				},
+			},
+		},
+		{
+			name: "regular privileged image, assigned to infra node, allow",
+			podSpec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Image: "regulardimage",
+						SecurityContext: &corev1.SecurityContext{
+							Privileged: &[]bool{true}[0],
+						},
+					},
+				},
+				NodeName: "infra-123456-000002",
+			},
+			response: &admissionv1beta1.AdmissionResponse{
+				UID:     "uid",
+				Allowed: true,
+				Result: &metav1.Status{
+					Status: metav1.StatusSuccess,
+				},
+			},
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			pod, err := json.Marshal(&corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+				},
+				Spec: test.podSpec,
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			req, err := json.Marshal(&admissionv1beta1.AdmissionReview{
+				Request: &admissionv1beta1.AdmissionRequest{
+					UID:      "uid",
+					Kind:     metav1.GroupVersionKind{Version: "v1", Kind: "Pod"},
+					Resource: metav1.GroupVersionResource{Version: "v1", Resource: "pods"},
+					Object: runtime.RawExtension{
+						Raw: pod,
+					},
+				}})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			//log.Printf("%s", string(req))
+
+			r := &http.Request{
+				Method: http.MethodPost,
+				Header: http.Header{"Content-Type": []string{"application/json"}},
+				Body:   ioutil.NopCloser(bytes.NewReader(req)),
+			}
+
+			w := newFakeResponseWriter()
+
+			ac.handleWhitelist(w, r)
+
+			if w.statusCode != 200 {
+				t.Errorf("got status code %d, %s", w.statusCode, w.Buffer.String())
+			}
+			if !reflect.DeepEqual(w.Header(), http.Header{"Content-Type": []string{"application/json"}}) {
+				t.Errorf("got header %#v", w.Header())
+			}
+
+			var rev *admissionv1beta1.AdmissionReview
+			err = json.NewDecoder(w).Decode(&rev)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(rev.Response, test.response) {
+				t.Errorf("got respose %#v", rev.Response)
+			}
+		})
+	}
+}
+
+func TestHandleSCCHappyPath(t *testing.T) {
+	client := fake.NewSimpleClientset(&core.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+			Annotations: map[string]string{
+				"openshift.io/sa.scc.uid-range": "1000/10",
+				"openshift.io/sa.scc.mcs":       "mcs",
+			},
+		},
+	})
+
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	ac := &admissionController{
+		client: client,
+		log:    logrus.NewEntry(logrus.StandardLogger()),
+	}
+	ac.protectedSCCs = ac.InitProtectedSCCs()
+
+	for _, test := range []struct {
+		name     string
+		scc      string
+		response *admissionv1beta1.AdmissionResponse
+	}{
+		{
+			name: "protected SCC, added user, allow",
+			scc: `{
+				"metadata": {
```

Comment:
+1 for the unit test, but its verbosity adds a maintenance overhead - can you find a way to simplify the boilerplate?